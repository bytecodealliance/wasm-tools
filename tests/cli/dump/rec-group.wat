;; RUN: dump %

(module
  (rec
    (type (func (param i32 i32) (result f64)))
  )
  (rec
    (type (sub (struct (field i32))))
    (type (sub (struct (field (mut i32)))))
    (type (sub (struct (field (mut i32)
                              (mut i64)
                              (mut f32)
                              (mut f64)
                              (mut v128)
                              (mut funcref)
                              (mut externref)
                              (mut (ref null 2))))))
  )
  (type $a (sub (array i32)))
  (type $b (sub $a (array i32)))
)
