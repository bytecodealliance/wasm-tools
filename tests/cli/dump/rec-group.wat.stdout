  0x0 | 00 61 73 6d | version 1 (Module)
      | 01 00 00 00
  0x8 | 01 37       | type section
  0xa | 04          | 4 count
--- rec group 0 (explicit) ---
  0xb | 4e 01       | 
  0xd | 60 02 7f 7f | [type 0] SubType { is_final: true, supertype_idx: None, composite_type: Func(FuncType { params: [I32, I32], results: [F64] }) }
      | 01 7c      
--- rec group 1 (explicit) ---
 0x13 | 4e 03       | 
 0x15 | 50 00 5f 01 | [type 1] SubType { is_final: false, supertype_idx: None, composite_type: Struct(StructType { fields: [FieldType { element_type: Val(I32), mutable: false }] }) }
      | 7f 00      
 0x1b | 50 00 5f 01 | [type 2] SubType { is_final: false, supertype_idx: None, composite_type: Struct(StructType { fields: [FieldType { element_type: Val(I32), mutable: true }] }) }
      | 7f 01      
 0x21 | 50 00 5f 08 | [type 3] SubType { is_final: false, supertype_idx: None, composite_type: Struct(StructType { fields: [FieldType { element_type: Val(I32), mutable: true }, FieldType { element_type: Val(I64), mutable: true }, FieldType { element_type: Val(F32), mutable: true }, FieldType { element_type: Val(F64), mutable: true }, FieldType { element_type: Val(V128), mutable: true }, FieldType { element_type: Val(Ref(funcref)), mutable: true }, FieldType { element_type: Val(Ref(externref)), mutable: true }, FieldType { element_type: Val(Ref((ref null (module 2)))), mutable: true }] }) }
      | 7f 01 7e 01
      | 7d 01 7c 01
      | 7b 01 70 01
      | 6f 01 63 02
      | 01         
--- rec group 2 (implicit) ---
 0x36 | 50 00 5e 7f | [type 4] SubType { is_final: false, supertype_idx: None, composite_type: Array(ArrayType(FieldType { element_type: Val(I32), mutable: false })) }
      | 00         
--- rec group 3 (implicit) ---
 0x3b | 50 01 04 5e | [type 5] SubType { is_final: false, supertype_idx: Some(CoreTypeIndex { kind: "module", index: 4 }), composite_type: Array(ArrayType(FieldType { element_type: Val(I32), mutable: false })) }
      | 7f 00      
 0x41 | 00 0e       | custom section
 0x43 | 04 6e 61 6d | name: "name"
      | 65         
 0x48 | 04 07       | type name section
 0x4a | 02          | 2 count
 0x4b | 04 01 61    | Naming { index: 4, name: "a" }
 0x4e | 05 01 62    | Naming { index: 5, name: "b" }
